src/norm_flows.jl

```diff
@@ -0,0 +1,121 @@
+using Distributions
+using LinearAlgebra
+using Random
+using StatsFuns: softplus
+using Roots # for inverse
+
+################################################################################
+#                            Planar and Radial Flows                           #
+#             Ref: Variational Inference with Normalizing Flows,               #
+#               D. Rezende, S. Mohamed(2015) arXiv:1505.05770                  #
+################################################################################
+
+mutable struct PlanarLayer{T1,T2} <: Bijector
+    w::T1
+    u::T1
+    b::T2
+end
+
+(b::PlanarLayer)(x) = transform(b, x)
+
+function get_u_hat(u, w)
+    # To preserve invertibility
+    return (
+        u + (planar_flow_m(w' * u) - w' * u)[1]
+        * w / (norm(w[:,1],2) ^ 2)
+    )   # from A.1
+end
+
+function PlanarLayer(dims::Int, container=Array)
+    w = container(randn(dims, 1))
+    u = container(randn(dims, 1))
+    b = container(randn(1))
+    return PlanarLayer(w, u, b)
+end
+
+planar_flow_m(x) = -1 .+ softplus.(x)   # for planar flow from A.1
+dtanh(x) = 1 .- (tanh.(x)) .^ 2         # for planar flow
+ψ(z, w, b) = dtanh(w' * z .+ b) .* w    # for planar flow from eq(11)
+
+# An internal version of transform that returns intermediate variables
+function _transform(flow::PlanarLayer, z)
+    u_hat = get_u_hat(flow.u, flow.w)
+    transformed = z + u_hat * tanh.(flow.w' * z .+ flow.b) # from eq(10)
+    return (transformed=transformed, u_hat=u_hat)
+end
+
+transform(flow::PlanarLayer, z) = _transform(flow, z).transformed
+
+function forward(flow::T, z) where {T<:PlanarLayer}
+    transformed, u_hat = _transform(flow, z)
+    # Compute log_det_jacobian
+    psi = ψ(z, flow.w, flow.b)
+    log_det_jacobian = log.(abs.(1.0 .+ psi' * u_hat))      # from eq(12)
+    return (rv=transformed, logabsdetjac=log_det_jacobian)  # from eq(10)
+end
+
+function inv(flow::PlanarLayer, y)
```

Comment:
Is there a reason why we cannot do this using `Inverted{<: PlanarLayer}`? Seems to me that we could just implement this as
```julia
function (ib::Inversed{<: PlanarLayer})(y)
    flow = ib.orig
    # same as in this inv(flow, y) ...
end
```